sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -477,6 +545,20 @@ impl TransportWrapper {
         self.transport.capabilities()
     }
 
+    /// Returns a string->string map containing user-defined aspects "provided" by the testbed
+    /// setup.  For instance, whether a SPI flash chip is fitted in the socket, or whether pullup
+    /// resistors are suitable for high-speed I2C.
+    pub fn provides_map(&self) -> Result<&HashMap<String, String>> {
+        Ok(&self.provides_map)
+    }
+
+    pub fn query_provides(&self, key: &str) -> Result<&str> {
+        self.provides_map.get(key).map(|x| &**x).ok_or_else(|| {
```

Comment:
The map expression could probably be `map(String::as_str)`.

Reply:
That looks much better.

I kept trying all sorts of `.as_deref()` on the option itself, hoping to be able to convert a `Option<String>` to `Option<&str>`, without success, and finally did the horrible `map()` that the rust compiler suggested in its error message.

Reply:
It seems like `as_deref()` should work here.  I think the reason it doesn't is because the type returned by `get()` is `Option<&String>` and not `Option<String>`.  Internally, `Option::as_deref` uses `as_ref` on itself, which creates an `Option<&&String>`, which is then dereffed back to `&String`.